internal/metrics/metrics.go

```diff
@@ -127,10 +128,15 @@ func ExtractFunctionsFromCallTrees[T comparable](
 	callTrees map[T][]*nodetree.Node,
 ) []nodetree.CallTreeFunction {
 	functions := make(map[uint32]nodetree.CallTreeFunction, 0)
-
-	for _, callTreesForThread := range callTrees {
+	threadID := ""
+	for tid, callTreesForThread := range callTrees {
+		if t, ok := any(tid).(string); ok {
+			threadID = t
+		} else if t, ok := any(tid).(uint64); ok {
+			threadID = strconv.FormatUint(t, 10)
+		}
```

Comment:
The initialization of `threadID`should happen inside the loop as this risks the possibility that the thread id from the previous iteration is carried over into the next.

```suggestion
	for tid, callTreesForThread := range callTrees {
		var threadID string
		if t, ok := any(tid).(string); ok {
			threadID = t
		} else if t, ok := any(tid).(uint64); ok {
			threadID = strconv.FormatUint(t, 10)
		}
```